batch/src/main/java/org/openmrs/analytics/JdbcFhirMode.java

```diff
@@ -0,0 +1,236 @@
+			                this.tableName, tableId, tableId))
+			            .withRowMapper(new JdbcIO.RowMapper<String>() {
+				            
+				            @Override
+				            public String mapRow(ResultSet resultSet) throws Exception {
+					            return resultSet.getString("uuids");
+				            }
+			            }).withCoder(NullableCoder.of(StringUtf8Coder.of())));
+		}
+	}
+	
+	public static class CreateSearchSegments extends PTransform<PCollection<String>, PCollection<SearchSegmentDescriptor>> {
+		
+		private final String baseBundleUrl;
+		
+		private final int parallelRequests;
+		
+		public CreateSearchSegments(String baseBundleUrl, int parallelRequests) {
+			this.baseBundleUrl = baseBundleUrl;
+			this.parallelRequests = parallelRequests;
+		}
+		
+		@Override
+		public PCollection<SearchSegmentDescriptor> expand(PCollection<String> uuIds) {
+			return uuIds
+			        // create KV required by GroupIntoBatches
+			        .apply(ParDo.of(new DoFn<String, KV<Integer, String>>() {
+				        
+				        @ProcessElement
+				        public void processElement(ProcessContext c) {
+					        if (c.element() != null) {
+						        c.output(KV.of(c.element().split(",").length,
+						            String.format("%s?_id=%s", baseBundleUrl, c.element())));
+					        }
+				        }
+			        }))
+			        // batchSize is used here instead of jdbcFetchSize to limit the number of  `_id` passed per request
+			        // TODO: find out why passing batchSize > 1000 sometimes results in OpenMRS internal server error
+			        .apply(String.format("GroupIntoBatches of %s", this.parallelRequests),
+			            GroupIntoBatches.<Integer, String> ofSize(this.parallelRequests))
+			        .apply("Generate Segments", ParDo.of(new DoFn<KV<Integer, Iterable<String>>, SearchSegmentDescriptor>() {
+				        
+				        @ProcessElement
+				        public void processElement(@Element KV<Integer, Iterable<String>> element,
+				                OutputReceiver<SearchSegmentDescriptor> r) {
+					        element.getValue().forEach(
+					            bundleUrl -> r.output(SearchSegmentDescriptor.create(bundleUrl, element.getKey())));
+					        
+				        }
+			        }));
+		}
+	}
+	
+	public JdbcIO.DataSourceConfiguration getJdbcConfig(String jdbcDriverClass, String jdbcUrl, String dbUser,
+	        String dbPassword, int dbcMaxPoolSize) throws PropertyVetoException {
+		ComboPooledDataSource dataSource = new ComboPooledDataSource();
+		dataSource.setDriverClass(jdbcDriverClass);
+		dataSource.setJdbcUrl(jdbcUrl);
+		dataSource.setUser(dbUser);
+		dataSource.setPassword(dbPassword);
+		dataSource.setMaxPoolSize(dbcMaxPoolSize);
+		dataSource.setInitialPoolSize(10); // TODO: make this configurable if issues arises
+		return JdbcIO.DataSourceConfiguration.create(dataSource);
+	}
+	
+	public LinkedHashMap<String, String> createFhirReverseMap(String searchString, String tableFhirMapPath)
```

Comment:
Is there a particular reason that we need a `LinkedHashMap` to be returned from this instead of a simple `Map` interface?

Reply:
No reason at all, I have update to return Map instead of LinkedHashMap! Thanks @bashir2